primitives/xcm-transactor/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use codec::{Decode, Encode, FullCodec};
+pub use cumulus_primitives_core::ParaId;
+use frame_support::RuntimeDebug;
+use xcm::{latest::Weight as XcmWeight, prelude::*};
+
+pub trait BuildRelayCall {
+	type RelayCall: FullCodec;
+	/// TODO: Add Notes about which function this is and why it is being used
+	///
+	///
+	fn swap_call(self_para_id: ParaId, other_para_id: ParaId) -> Self::RelayCall;
+
+	/// Wraps constructed Relaychain Call in an XCM message to be dispatched via 'send_xcm'
+	///
+	/// Params:
+	/// - RelayCall (Different depending on Kusama or Polkadot)
+	/// - execution to be purchased via BuyExecution Xcm Instruction
+	/// - Weight required to execute this call.
+	///
+	/// Returns:
+	/// - Corresponding Xcm Message for Transacting on this RelayCall
+	///
+	fn construct_transact_xcm(call: Self::RelayCall, weight: XcmWeight) -> Xcm<()>;
+}
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum RegistrarCall {
+	/// Corresponds to the swap extrinsic index within the Registrar Pallet
+	#[codec(index = 3)]
+	Swap(ParaId, ParaId),
```

Comment:
I prefer named fields when it comes to such enum args, otherwise it is very unclear, e.g.:

```rust
pub enum RegistrarCall {

    Swap {
       this: ParaId
       other: ParaId 
    }
}
```